photon-client/src/integTest/scala/com/linkedin/photon/ml/data/DataValidatorsTest.scala

```diff
@@ -51,94 +53,188 @@ class DataValidatorsTest extends SparkTestUtils {
+    val validVector = vectors.head
+    val invalidVector = vectors.last
+
+    // labeled points with valid vectors
+    val lpPositiveLabel = new LabeledPoint(5.0, validVector)
+    val lpNegativeLabel = new LabeledPoint(-5.0, validVector)
+    val lpBinaryLabel = new LabeledPoint(BinaryClassifier.negativeClassLabel, validVector)
+
+    // labeled point with invalid label
+    val lpInfLabel = new LabeledPoint(Double.PositiveInfinity, validVector)
+
+    // labeled point with invalid offset
+    val lpInfOffset = new LabeledPoint(BinaryClassifier.positiveClassLabel, validVector, Double.NaN)
+
+    // labeled points with invalid vectors
+    val lpNonBinaryLabelInfFeatures = new LabeledPoint(-2.0, invalidVector)
+    val lpBinaryLabelInfFeatures = new LabeledPoint(BinaryClassifier.negativeClassLabel, invalidVector)
+
+    Assert.assertNotNull(sc)
+
+    // All RDDs have one valid point and at least one invalid point
+    Array(
+      Array(
+        sc.parallelize(List(lpPositiveLabel, lpInfLabel)),
+        TaskType.LINEAR_REGRESSION,
+        DataValidationType.VALIDATE_FULL, false),
+      Array(
+        sc.parallelize(List(lpPositiveLabel, lpNonBinaryLabelInfFeatures)),
+        TaskType.LINEAR_REGRESSION,
+        DataValidationType.VALIDATE_FULL, false),
+      Array(
+        sc.parallelize(List(lpPositiveLabel, lpInfOffset)),
+        TaskType.LINEAR_REGRESSION,
+        DataValidationType.VALIDATE_FULL, false),
+
+      Array(
+        sc.parallelize(List(lpBinaryLabel, lpPositiveLabel)),
+        TaskType.LOGISTIC_REGRESSION,
+        DataValidationType.VALIDATE_FULL, false),
+      Array(
+        sc.parallelize(List(lpBinaryLabel, lpInfLabel)),
+        TaskType.LOGISTIC_REGRESSION,
+        DataValidationType.VALIDATE_FULL, false),
+      Array(
+        sc.parallelize(List(lpBinaryLabel, lpBinaryLabelInfFeatures)),
+        TaskType.LOGISTIC_REGRESSION,
+        DataValidationType.VALIDATE_FULL, false),
+      Array(
+        sc.parallelize(List(lpBinaryLabel, lpInfOffset)),
+        TaskType.LOGISTIC_REGRESSION,
+        DataValidationType.VALIDATE_FULL, false),
+
+      Array(
+        sc.parallelize(List(lpPositiveLabel, lpInfLabel)),
+        TaskType.POISSON_REGRESSION,
+        DataValidationType.VALIDATE_FULL, false),
+      Array(
+        sc.parallelize(List(lpPositiveLabel, lpNegativeLabel)),
+        TaskType.POISSON_REGRESSION,
+        DataValidationType.VALIDATE_FULL, false),
+      Array(
+        sc.parallelize(List(lpPositiveLabel, lpNonBinaryLabelInfFeatures)),
+        TaskType.POISSON_REGRESSION,
+        DataValidationType.VALIDATE_FULL, false),
+      Array(
+        sc.parallelize(List(lpPositiveLabel, lpInfOffset)),
+        TaskType.POISSON_REGRESSION,
+        DataValidationType.VALIDATE_FULL, false)
+    )
   }
 
-  /*
   // TODO This test does not work because the spark context ends up being null in the data provider. Unlike other tests
```

Comment:
Yes - I wanted to keep the comment to explain why the tests call the data provider differently than they're supposed to, but forgot that it references a specific test.